internal/extender/resource.go

```diff
@@ -120,6 +131,16 @@ func failWithMessage(ctx context.Context, args schedulerapi.ExtenderArgs, messag
 	return &schedulerapi.ExtenderFilterResult{FailedNodes: failedNodes}
 }
 
+func (s *SparkSchedulerExtender) reconcileIfNeeded(ctx context.Context) error {
+	now := time.Now()
+	var err error
+	if s.lastRequest.Add(leaderElectionInterval).After(now) {
+		err = s.syncResourceReservationsAndDemands(ctx)
+	}
+	s.lastRequest = now
```

Comment:
I think you only want to set this if err != nil

Reply:
done